src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -244,6 +248,9 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                 _watcherRunLoop = Interop.RunLoop.CFRunLoopGetCurrent();
                 Debug.Assert(_watcherRunLoop != IntPtr.Zero);
 
+                // Retain the RunLoop so that it doesn't get moved or cleaned up before we're done with it.
+                Interop.CoreFoundation.CFRetain(_watcherRunLoop);
```

Comment:
Thanks, good suggestion. I was on the fence about whether to set _watcherRunLoop to the result but didn't see the point. This is better.